components/camel-as2/camel-as2-api/src/main/java/org/apache/camel/component/as2/api/AS2ClientConnection.java

```diff
@@ -67,12 +84,16 @@ public AS2ClientConnection(String as2Version, String userAgent, String clientFqd
                 .add(new RequestConnControl())
                 .add(new RequestExpectContinue(true)).build();
 
-        // Create Socket
-        Socket socket = new Socket(targetHost.getHostName(), targetHost.getPort());
+        this.httpRoute = new HttpRoute(new HttpHost(targetHost.getHostName(), targetHost.getPort()));
 
-        // Create Connection
-        httpConnection = new AS2BHttpClientConnection(8 * 1024);
-        httpConnection.bind(socket);
+        org.apache.http.conn.HttpConnectionFactory<HttpRoute, ManagedHttpClientConnection> httpConnectionFactory = new HttpConnectionFactory<HttpRoute, ManagedHttpClientConnection>() {
```

Comment:
I am bit confused here, is `httpConnectionFactory` supposed to be initialized but never used somewhere else?